exercises/run-length-encoding/src/example.cr

```diff
@@ -0,0 +1,40 @@
+module RunLengthEncoding
+  extend self
+
+  def encode(str)
+    reader = Char::Reader.new(str)
+    String.build do |io|
+      while reader.has_next?
+        count = 1
+
+        while reader.current_char == reader.peek_next_char
+          count += 1
+          reader.next_char
+        end
+
+        io << count if count > 1
+        io << reader.current_char
+        reader.next_char
+      end
+    end
+  end
+
+  def decode(str)
+    reader = Char::Reader.new(str)
```

Comment:
I'm wondering if this section could be DRY'd out but... it might be more trouble than it's worth...
```crystal
reader = Char::Reader.new(str)
String.build do |io|
  while reader.has_next?
  end
end
```